### DISCUSSION TITLE
Slint lsp for Neovim not support on MacOS

### DISCUSSION THREAD

**Discussion by Grieey** [slint-ui/slint#2715] [Q&A]
I'm using Mason plugin to manage lsp.And it's said not support on MacOS, so is there any plan to support it?
![image](https://github.com/slint-ui/slint/assets/49085504/7b1fe892-fd11-459c-b234-90dafb8801d3)

**Comment by ogoffart** [comment]
MacOS should be supported. This error message doesn't come from Slint

**Reply by ogoffart** [reply]
I think I understand now where the message comes from. Probably related to the fact that we don't have binary for MacOS?

**Reply by tronical** [reply]
@hunger do you know what binaries might be involved here?

**Reply by ogoffart** [reply]
I think I remember @hunger said something like the plugin would download the binaries from github release.  But I couldn't find where this happens with a quick search.

**Reply by hunger** [reply]
I added some code into the [`nvim-lsp-installer`](https://github.com/williamboman/nvim-lsp-installer) plugin to download slint-lsp from github and to make it available to neovim. [Mason](https://github.com/williamboman/mason.nvim) seems to have replaced nvim-lsp-installer in the meantime.

This downloader plugin works great on Linux and Windows, but there is no stand-alone language server binary for Mac at this time, so it fails there. IIRC building the language server for Mac requires code signing to be set up on github and we never got around to set that up.

**Reply by tronical** [reply]
Ah, @hunger, do you mean that the installer downloads the binaries from https://github.com/slint-ui/slint/releases ?

**Reply by hunger** [reply]
@tronical: Yes, it downloads straight from the github release page you linked.

**Reply by tronical** [reply]
ACK. That explains it.

**Comment by lynndylanhurley** [comment]
I pushed a PR here that might resolve this:
https://github.com/mason-org/mason-registry/pull/3740
